docs/decisions/app/0005-server-rendering.md

```diff
@@ -0,0 +1,73 @@
+
+If a project team is building a site that renders the same content for every user, they can change their application to utilize [Next.js's static HTML export functionality](https://nextjs.org/docs/advanced-features/static-html-export) (prerendering). This flexibility to do either server rendering or prerendering within the same React framework is one benefit to using Next.js.
+
+## Pros and Cons of the Options
+
+### Server rendering
+
+Pros
+
+- Data fetching occurs on the server. The browser natively handles the page's loading state. This means less overall code to write, test, and maintain. An uncaught error on the server will be louder (in a good way) than an uncaught error on the client (which could result in a never ending spinner).
+- Makes it easier to implement a clearer separation of concerns:
+  - [Middleware](https://nextjs.org/docs/advanced-features/middleware&sa=D&source=editors&ust=1675374865787476&usg=AOvVaw3zDFoDM50pSoew_4SsZpjm) provide a place for enforcing auth, reading/setting secure cookies, setting HTTP headers, and redirects.
+  - [Loaders](https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props&sa=D&source=editors&ust=1675374865787877&usg=AOvVaw0zAh7tsjHKl2P8pDGVhf-0) provide a place for fetching all data required for rendering the page.
+- Running page logic and rendering on the server makes it possible to send lighter payloads to the client. This approach can work well for a large spectrum of device and network conditions. [You can make your server fast, but you can't control the user's device or network](https://remix.run/docs/en/v1/pages/philosophy%23serverclient-model&sa=D&source=editors&ust=1675374865788271&usg=AOvVaw3bPMcx9gLfFygQu_qL2R-9).
+- Data fetching on the server enables accessing authenticated APIs (e.g. using [TLS mutual auth](https://www.cloudflare.com/learning/access-management/what-is-mutual-tls/) to talk to fetch data from a protected third-party API).
+- Low effort to implement pretty [dynamic routes](https://nextjs.org/docs/routing/dynamic-routes&sa=D&source=editors&ust=1675374865788677&usg=AOvVaw070KKTAMkbV2YkqYVRpep-) (e.g `/claim/:claim_id`)
+- [API routes](https://nextjs.org/docs/api-routes/introduction) can be created to handle other types of HTTP requests (POST, PUT, etc).
+- Nice side benefit: Server rendering is the only option for [Remix](https://remix.run/). It may be easier, from an infra standpoint and as a conceptual model, to migrate to Remix if the Next.js apps we're building were server rendered.
+
+Cons
+
+- Requires infra resources to run the containerized application, and all the things that come along with a server (rate limiting, auto scaling), such as [AWS App Runner](https://aws.amazon.com/apprunner/). This can have higher costs than a prerendered site, and requires more effort to receive security approvals due to a larger attack surface.
```

Comment:
I think this is missing the biggest practical Con, which the operational + compliance burden of maintaining a server:

* Security audits easily take 2-3 months of multiple engineers time, and producing the documentation for those audits is pretty onerous
* Having a server generally requires a 24/7 on-call rotation since servers are succeptable to outages

Reply:
> Having a server generally requires a 24/7 on-call rotation since servers are succeptable to outages

Isn't this true for static too? PFML Portal was static and it seemed like the expectation for Portal on-call engineers was that it was 24/7.

Reply:
> Having a server generally requires a 24/7 on-call rotation

I'm wondering if this would apply to any site of importance, regardless of whether the site is static or not. Even for static sites, you have the cloud provider or downstream API as a failure point.

Reply:
> I'm wondering if this would apply to any site of importance, regardless of whether the site is static or not. Even for static sites, you have the cloud provider or downstream API as a failure point.

that's true, maybe it's more of the fact that the probability of certain categories of incidents goes almost to zero, so the amount of burden of being on-call is dramatically reduced. it's like going from 99% availability of ECS to 99.99% availability of cloudfront/s3 (i made up those numbers).

downstream API failure might be true, but if that's a separate component then that's a separate on-call team.